### PR TITLE
Re-order Build Tasks to Publish to the Ballerina Central

### PR DESCRIPTION
**Pull request by Maninda**
$subject.

### CODE REVIEW

**Review by ThisaruGuruge — COMMENTED** [review]
Let's remove the testable jar related changes

File: email-ballerina/build.gradle [hunk 1]
```diff
@@ -28,6 +28,7 @@ def artifactBallerinaDocs = file("$project.projectDir/build/docs_parent/")
 def artifactCacheParent = file("$project.projectDir/build/cache_parent/")
 def artifactLibParent = file("$project.projectDir/build/lib_parent/")
 def artifactCodeCoverageReport = file("$project.projectDir/target/cache/tests_cache/coverage/ballerina.exec")
+def artifactTestableJar = file("$project.projectDir/target/cache/${packageOrg}/${packageName}/${tomlVersion}/java11/")
```
Comment (Maninda) [inline email-ballerina/build.gradle:31]:
```suggestion
```

File: email-ballerina/build.gradle [hunk 2]
```diff
@@ -282,6 +288,14 @@ task createArtifactZip(type: Zip) {
...
 }
 
+task createTestableJarZip(type: Zip) {
+    archiveName("${packageOrg}-${packageName}-${project.version}-testable-jars.zip")
+    destinationDir(file("${buildDir}/distributions"))
+    from("${artifactTestableJar}") {
+        include '**/*-testable.jar'
+    }
+}
+
```
Comment (Maninda) [inline email-ballerina/build.gradle:298]:
```suggestion
```

File: email-ballerina/build.gradle [hunk 3]
```diff
@@ -306,14 +320,32 @@ publishing {
...
+    }
+
+    jars.forEach { file ->
+        from zipTree(file).matching {
+            exclude '**/tests/*'
+            include '**/*.class'
+        }
+        into "${project.rootDir.absolutePath}/build/classes"
+    }
+}
```
Comment (Maninda) [inline email-ballerina/build.gradle:335]:
```suggestion
```

File: email-ballerina/build.gradle [hunk 4]
```diff
@@ -306,14 +313,19 @@ publishing {
     }
 }
 
+
```
Comment (Maninda) [inline email-ballerina/build.gradle:316]:
```suggestion
```
Remove extra empty line.

File: email-ballerina/build.gradle [hunk 5]
```diff
@@ -170,6 +174,7 @@ task initializeVariables {
 
         if (graph.hasTask(":${packageName}-ballerina:publish")) {
             ballerinaPublish.enabled = true
+            ballerinaTest.enabled = false
```
Comment (ThisaruGuruge) [inline email-ballerina/build.gradle:177]:
```suggestion
```

**Review by ThisaruGuruge — APPROVED** [review]

**Pull request merged**